source/devoptab/devoptab_fsa_utils.cpp

```diff
@@ -0,0 +1,169 @@
+#include "devoptab_fs.h"
+
+char *
+__fsa_fixpath(struct _reent *r,
+              const char *path) {
+    char *p;
+    char *fixedPath;
+
+    if (!path) {
+        r->_errno = EINVAL;
+        return nullptr;
+    }
+
+    p = strchr(path, ':') + 1;
+    if (!strchr(path, ':')) {
+        p = (char *) path;
+    }
+
+    if (strlen(p) > FS_MAX_PATH) {
+        r->_errno = ENAMETOOLONG;
+        return nullptr;
+    }
```

Comment:
```suggestion
    size_t pathLength = strlen(p);
    if (pathLength > FS_MAX_PATH) {
        r->_errno = ENAMETOOLONG;
        return nullptr;
    }
    
    // Wii U softlocks on empty paths so return proper error code
    if (pathLength  == 0) {
        r->_errno = ENOENT;
        return nullptr;
    }
```
Preemptively return error on empty paths since the Wii U softlocks. Got a crash with libmocha when I was creating directories recursively.